src/lsm/superblock_free_set.zig

```diff
@@ -175,12 +188,32 @@ pub const SuperBlockFreeSet = struct {
     }
 
     /// Returns the number of bytes written to `target`.
+    /// The encoded data does *not* incldue staged changes.
     pub fn encode(set: SuperBlockFreeSet, target: []align(@alignOf(usize)) u8) usize {
         assert(target.len == SuperBlockFreeSet.encode_size_max(set.blocks.bit_length));
 
         return ewah.encode(bitset_masks(set.blocks), target);
     }
 
+    /// Returns the number of bytes written to `target`.
+    /// The encoded data *does* include staged changes.
+    pub fn encode_with_staging(set: *SuperBlockFreeSet, target: []align(@alignOf(usize)) u8) usize {
+        const count_free = set.count_released();
+        const count_staged = set.staging.count();
+        assert(target.len == SuperBlockFreeSet.encode_size_max(set.blocks.bit_length));
```

Comment:
This assert on the target arg could be moved earlier right up to the top of the function, since it's a pre-condition assert and doesn't depend on `count_free/staged` I think.

Reply:
It would bring the counts also a fraction closer to where we use them.